controllers/infinispan_controller.go

```diff
@@ -290,6 +290,37 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 		}
 	}
 
+	if infinispan.HasSites() {
```

Comment:
If someone remove `sites` section from the CR, Deployment will keep forever. Seems we need validate it and cleanup if needed or set `sites` section as immutable in the webhook controller (not implemented now)

Reply:
where can it be set immutable? 

Removing the deployment wouldn't help since JGroups and Infinispan are still configured to use cross-site replication. I guess it would be easier to set the field immutable than restarting and reconfiguring the cluster. 
Do you agree?

Reply:
Seem to if someone remove `sites` structure, configuration ConfgiMap will be changed and StatefulSet will be rollout to the new config without cross-site setup. Is'n it?

Reply:
I'll test this locally to see what the operator does and acting accordingly. thanks!

Reply:
@dmvolod you are right! 
I've updated the code to remove the `-site` `service` and the `deployment`.